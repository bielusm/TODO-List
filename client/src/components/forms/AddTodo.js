import React, { useState } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Card,
  Form,
  FormGroup,
  Label,
  Input,
  Button,
  CardHeader,
  CardBody
} from 'reactstrap';
import { addTodo } from '../../actions/todo';

export const AddTodo = ({ addTodo, history }) => {
  const [formData, setFormData] = useState({
    name: '',
    description: ''
  });

  const { name, description } = formData;

  const onChange = e => {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  };

  const onSubmit = e => {
    e.preventDefault();
    addTodo(formData);
    history.push('/dashboard');
  };

  return (
    <Card>
      <CardHeader tag="h4">Add Todo</CardHeader>
      <CardBody>
        <Form onSubmit={e => onSubmit(e)}>
          <FormGroup>
            <Label for="name">Name</Label>
            <Input
              type="text"
              name="name"
              required
              value={name}
              onChange={e => onChange(e)}
              className="name"
            ></Input>
          </FormGroup>
          <FormGroup>
            <Label for="description">Description</Label>
            <Input
              type="textarea"
              name="description"
              value={description}
              className="description"
              onChange={e => onChange(e)}
            ></Input>
          </FormGroup>
          <Button color="primary">Add Todo</Button>
        </Form>
      </CardBody>
    </Card>
  );
};

AddTodo.propTypes = {
  addTodo: PropTypes.func.isRequired
};

export default connect(null, { addTodo })(withRouter(AddTodo));
