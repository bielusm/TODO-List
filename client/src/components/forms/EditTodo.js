import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import {
  Card,
  Form,
  FormGroup,
  Label,
  Input,
  Button,
  CardHeader,
  CardBody
} from 'reactstrap';
import { editTodoAction, setTodoAction, removeTodo } from '../../actions/todo';

export const EditTodo = ({
  removeTodo,
  setTodoAction,
  editTodoAction,
  history,
  todo,
  match
}) => {
  const [formData, setFormData] = useState({
    name: todo ? todo.name : '',
    description: todo ? todo.description : ''
  });

  useEffect(() => {
    removeTodo();

    setTodoAction(match.params.todoId);
  }, []);

  useEffect(() => {
    setFormData({
      name: todo ? todo.name : '',
      description: todo ? todo.description : ''
    });
  }, [todo]);

  const { name, description } = formData;

  const onChange = e => {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  };

  const onSubmit = e => {
    e.preventDefault();
    editTodoAction(todo._id, formData);
    history.push('/dashboard');
  };

  return (
    <Card>
      <CardHeader tag="h4">Edit Todo</CardHeader>
      <CardBody>
        <Form onSubmit={e => onSubmit(e)}>
          <FormGroup>
            <Label for="name">Name</Label>
            <Input
              id="name"
              type="text"
              name="name"
              required
              value={name}
              onChange={e => onChange(e)}
              className="name"
            ></Input>
          </FormGroup>
          <FormGroup>
            <Label for="description">Description</Label>
            <Input
              id="description"
              type="textarea"
              name="description"
              value={description}
              className="description"
              onChange={e => onChange(e)}
            ></Input>
          </FormGroup>
          <Button color="primary">Edit Todo</Button>
        </Form>
      </CardBody>
    </Card>
  );
};

EditTodo.propTypes = {
  editTodoAction: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  todo: state.todo.todo
});

export default connect(mapStateToProps, {
  editTodoAction,
  setTodoAction,
  removeTodo
})(withRouter(EditTodo));
