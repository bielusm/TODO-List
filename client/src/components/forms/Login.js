import React, { useState } from 'react';
import {
  Card,
  Form,
  FormGroup,
  Label,
  Input,
  Button,
  CardHeader,
  CardBody
} from 'reactstrap';

const Login = props => {
  const [formData, setFormData] = useState({
    email: '',
    password: ''
  });

  const onChange = e => {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  };

  const { email, password } = formData;

  return (
    <Card>
      <CardHeader tag="h4">Login</CardHeader>
      <CardBody>
        <Form>
          <FormGroup>
            <Label for="email">Email</Label>
            <Input
              type="email"
              name="email"
              value={email}
              onChange={e => onChange(e)}
            ></Input>
          </FormGroup>
          <FormGroup>
            <Label for="password">Password</Label>
            <Input
              type="password"
              name="password"
              value={password}
              onChange={e => onChange(e)}
            ></Input>
          </FormGroup>
          <Button color="primary">Submit</Button>
        </Form>
      </CardBody>
    </Card>
  );
};

export default Login;
