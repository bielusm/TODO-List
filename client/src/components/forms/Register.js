import PropTypes from 'prop-types';
import React, { useState } from 'react';
import {
  Card,
  Form,
  FormGroup,
  Label,
  Input,
  Button,
  CardHeader,
  CardBody
} from 'reactstrap';
import { Redirect } from 'react-router-dom';
import { register } from '../../actions/auth';
import { setAlert } from '../../actions/alert';
import { connect } from 'react-redux';

const Register = ({ authenticated, register, setAlert }) => {
  const [formData, setFormData] = useState({
    email: '',
    password: '',
    password2: ''
  });

  const { email, password, password2 } = formData;

  const onChange = e => {
    setFormData({ ...formData, [e.target.name]: e.target.value });
  };

  const onSubmit = e => {
    e.preventDefault();
    if (password === password2) register(formData);
    else setAlert('Passwords must match', 'danger');
  };

  if (authenticated) return <Redirect to="/dashboard" />;

  return (
    <>
      <Card>
        <CardHeader tag="h4" className="notepadTop">
          Register
        </CardHeader>
        <CardBody className="notepadBottom">
          <Form onSubmit={e => onSubmit(e)}>
            <FormGroup>
              <Label for="email">Email</Label>
              <Input
                type="email"
                name="email"
                required
                value={email}
                onChange={e => onChange(e)}
                className="notepadInput"
              ></Input>
            </FormGroup>
            <FormGroup>
              <Label for="password">Password</Label>
              <Input
                type="password"
                name="password"
                required
                value={password}
                onChange={e => onChange(e)}
                className="notepadInput"
              ></Input>
            </FormGroup>
            <FormGroup>
              <Label for="password2">Confirm Password</Label>
              <Input
                type="password"
                name="password2"
                required
                value={password2}
                onChange={e => onChange(e)}
                className="notepadInput"
              ></Input>
            </FormGroup>
            <Button color="primary" className="notepadBtn">
              Submit
            </Button>
          </Form>
        </CardBody>
      </Card>
    </>
  );
};

const mapStateToProps = state => ({
  authenticated: state.user.authenticated
});

Register.propTypes = {
  register: PropTypes.func.isRequired,
  setAlert: PropTypes.func.isRequired,
  authenticated: PropTypes.bool.isRequired
};

export default connect(mapStateToProps, { register, setAlert })(Register);
