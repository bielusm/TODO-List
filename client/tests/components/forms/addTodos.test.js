import React from 'react';
import { shallow } from 'enzyme';
import { AddTodo } from '../../../src/components/forms/AddTodo';
test('snapshot add todo form', () => {
  let wrapper = shallow(<AddTodo authenticated={true} />);
  expect(wrapper).toMatchSnapshot();

  wrapper = shallow(<AddTodo authenticated={false} />);
  expect(wrapper).toMatchSnapshot();
});
