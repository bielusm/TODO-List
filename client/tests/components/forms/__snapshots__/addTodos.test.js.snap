// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot add todo form 1`] = `ShallowWrapper {}`;

exports[`snapshot add todo form 2`] = `ShallowWrapper {}`;
